receivers-in-go.md

```diff
@@ -0,0 +1,257 @@
+# Neither self nor this: Receivers in Go
+
+When getting started with Go, there is a strong temptation to bring baggage from
+your previous language. It’s a heuristic which is usually helpful, but sometimes
+counter-productive and inevitably results in regret.
+
+Go does not have classes and objects, but it does have types that we can make
+many instances of. Further, we can attach methods to these types and they
+kind-of start looking like the classes we’re used to. When we attach a method
+to a type, the receiver is the instance of the type for which it was called.
+
+
+## Quick refresher on Go structs
+
+```go
+type Server struct {
+    ...
+}
+
+func (srv *Server) Close() error {
+    ...
+}
+
+func (srv Server) Name() string {
+    ...
+}
+```
+
+Every time I start writing somekind of server, it starts out looking like this.  
+It's a type that holds information about our server, like a `net.Listener` 
+socket, and it has a `Close()` method that shuts down the server. Easy enough.
+
+The *receiver* of the `Close()` method is the `(srv *Server)` part. This says 
+that inside of the `Close()` method declaration, the scope will have an `srv` 
+variable that is a reference to the instance of the `Server` that it's being 
+called on.  That is:
+
+```go
+myServer := &Server{}
+myServer.Close()
+```
+
+In this case, the `s` that is referenced inside of the `myServer.Close()` is 
+effectively the same variable as `myServer`. They're both references to the same 
+`Server` instance.
+
+
+## Naming the receiver
+
+In a lot of Go code, it's common to use the first letter or a short abbreviation 
+as the name of the receiver. If the name of the struct is `Server`, we'll 
+usually see `s` or `srv` or even `server`. All of these are fine.
+
+But why not `self` or `this`? Coming from Python, or Ruby, or JavaScript, it's 
+tempting to do something like:
+
+```go
+func (this *Server) Close() error {
+    ...
+}
+```
+
+That's one less decision to make every time we declare a struct. *All* of our 
+methods could use the same receiver. Any time we see `this` in the code, we'll 
+*know* that we're talking about the receiver, not some random local variable. It 
+will be GREAT!.. or will it?
```

Comment:
I think you could add a stronger hint as to why it won't work. Just something to get people's brains along the right path for what's to come.